### ISSUE TITLE
Problems with keyboard dragging

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#164]
Current implementation of keyboard dragging is problematic.  JavaScript doesn't naturally support multiple keystroke detection, so we have to do this ourselves.

This is handled now with a combination of keyup, and keydown events.  It works sort of like this:

``` js
      keydown: function( event ) {
        self.keyState[ event.keyCode || event.which ] = true;
      },
      keyup: function( event ) {
        self.keyState[ event.keyCode || event.which ] = false;
      }
```

A keystate is used to track which keys are down at any given time, and `step()` is used to update the model position property based on the key state.  So the Node will move for as long as a key is pressed down in milliseconds.

Pros:
- Provides natural keyboard dragging without keydown delay for press and hold behavior.

Cons:
- Some keyboards (like apple bluetooth) do not support keyup!
- Must be tied to step function

**Comment by jessegreenberg** [comment]
> Some keyboards (like apple bluetooth) do not support keyup!

This is the biggest problem.  How important is it that apple bluetooth keyboards are supported?

**Comment by jessegreenberg** [comment]
Testing, it looks like Apple bluetooth keyboards provide this on keyup:

> keyCode=0        which=0        charCode=0 

This makes it tough to track which key was released.

**Comment by jobara** [comment]
@jessegreenberg i tried this in iOS 10 and had the same issue.

**Comment by jobara** [comment]
@jessegreenberg I just tried pairing the bluetooth keyboard to my MBP I'm using macOS 10.12 and ran the sim in Safari. Using the bluetooth keyboard I was able to select the balloon and move it around with the WASD keys without the issue.

**Comment by jobara** [comment]
@jessegreenberg I tried with a non-apple bluetooth keyboard and experienced the same problem. Using an iPad.

**Comment by jobara** [comment]
If i can track down a camera connection kit for the iPad i'll try a usb keyboard but I'm not sure we have one of those adapters.

**Comment by jobara** [comment]
@jessegreenberg i tried this on an iPad using a USB Keyboard (non-apple) connected via the camera connection kit. It exhibited the same behaviour. I think this is not specific to Apple Bluetooth keyboards but to how iOS works with physical keyboards in general. Or maybe it has something to do with how they've implemented keyboard support because of their onscreen keyboard. Not sure.

**Comment by jessegreenberg** [comment]
Thanks for testing this @jobara.  I am going to close this issue.  We cannot support physical keyboards with iOS devices  because of this issues.  We can submit a bug report to apple which is going to be tracked in phetsims/a11y-issues#10.
